orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/tasks/EvaluateConditionTask.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.pipeline.tasks;
+
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.RetryableTask;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.conditions.ConditionConfigurationProperties;
+import com.netflix.spinnaker.orca.conditions.Condition;
+import com.netflix.spinnaker.orca.conditions.ConditionSupplier;
+import com.netflix.spinnaker.orca.pipeline.WaitForConditionStage.WaitForConditionContext;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nonnull;
+import java.time.Clock;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static com.netflix.spinnaker.orca.pipeline.WaitForConditionStage.WaitForConditionContext.*;
+
+@Component
+public class EvaluateConditionTask implements RetryableTask {
+  private static Logger log = LoggerFactory.getLogger(EvaluateConditionTask.class);
+  private final ConditionConfigurationProperties conditionsConfigurationProperties;
+  private final Optional<List<ConditionSupplier>> suppliers;
+  private final Clock clock;
+
+  @Autowired
+  public EvaluateConditionTask(
+    ConditionConfigurationProperties conditionsConfigurationProperties,
+    Optional<List<ConditionSupplier>> suppliers,
+    Clock clock
+  ) {
+    this.conditionsConfigurationProperties = conditionsConfigurationProperties;
+    this.suppliers = suppliers;
+    this.clock = clock;
+  }
+
+  @Override
+  public long getBackoffPeriod() {
+    return TimeUnit.MINUTES.toMillis(conditionsConfigurationProperties.getBackoffWaitMins());
+  }
+
+  @Override
+  public long getTimeout() {
+    return Integer.MAX_VALUE;
```

Comment:
this seems dangerous...? I feel like we should have some reasonable default and then let the user override in the UI

Reply:
Curious why you think this is dangerous. It's assumed that the default stage timeout will kick in once reached.
Happy to change it to a much smaller value. (This value was inspired by the waitStage)

Reply:
so maybe I am missing something, but i've been debugging something else related to timeouts and I have yet to find this mysterious default stage timeout :)
I am pretty sure pipelines generated by UI (without the manual "override timeout") will never have a default stage timeout (despite what deck makes you believe)
The waitStage has an an intrinsic timeout (the wait) so I feel like the `getTimeout` there is essentially the task timeout (though probably would be nicer to return the waitTime + some buffer to be safe).

Anyway, just seems that the timeouts are confusing and I think it's wrong for the task to never complete if the conditions aren't met, it should timeout eventually. If the there is a legit usecase to not timeout the user can override it in the pipeline definition

Reply:
Ok, if stage timeouts aren't respected that has got to be a bug somewhere. In any case, I will update the timeout here to be shorter to handle the situation of a near infinite wait. I think 2 hours is reasonable default.